### ISSUE TITLE
arm64 support?

### ISSUE THREAD

**Issue by filipef101** [hauxir/rapidbay#14]
Possible to?

**Comment by hauxir** [comment]
Definitely not an impossible task. If you can manage to build an arm64 docker image that works feel free to PR

**Comment by filipef101** [comment]
The problem is the base image hauxir/libtorrent-python3-ubuntu

**Comment by hauxir** [comment]
source for the base image: https://github.com/hauxir/libtorrent-ubuntu-python3

**Comment by filipef101** [comment]
awesome, was looking for that! I'd love to contribute to this repo, I wonder what is the best way to support arm64? I don't know much about docker. Would the best way to go for it is to use `buildx` to build the multiple archs? if so should it be for that base image, or should we remove the need for that base image and do it all in this repo's docker file?

If we just make that base image have multiple archs I guess it's just something you can do afaik

**Comment by hauxir** [comment]
I have no idea tbh, if you figure it out feel free to make a PR

**Comment by filipef101** [comment]
Yup, I think I got it, kind of just mixed both docker files, waiting for it to finish building (on the rpi so takes ages 😅 ).
If it works will publish it to the docker hub as rapidbayarm64 (?)

Also of topic, but any chance we could stream while downloading?

**Comment by hauxir** [comment]
There's a WIP branch for that on https://github.com/hauxir/rapidbay/pull/15

**Comment by filipef101** [comment]
fyi I gave up on this for this week at least, also because I wanted something that supports instant playback ala popcorntime, and currently looking at this first https://github.com/diericx/bevy

**Comment by fpor** [comment]
I attempted to build this on my pi3 ARMv7 but unfortunately it crashes when I compile libtorrent.
Wish it worked, regardless amazing project you got going on here. I found others but none with jackett as indexer which makes ur superior.

**Comment by hauxir** [comment]
@fpor @filipef101 

try using this as a base image:

hauxir/libtorrent-python3-ubuntu-arm64v8:latest

**Comment by fpor** [comment]
> 
> 
> @fpor @filipef101
> 
> try using this as a base image:
> 
> `hauxir/libtorrent-python3-ubuntu-arm64v8:latest`

Thanks for the reply, could you make an armhf/v7 version only pi4 is arm64.

**Comment by hauxir** [comment]
I unfortunately don't have a pi3 to build on. What is the issue you run into when you build the libtorrent-python3-ubuntu image there?
Have you tried using arm32v7/ubuntu:focal as a base image there?

**Comment by sunjam** [comment]
> > @fpor @filipef101
> > try using this as a base image:
> > `hauxir/libtorrent-python3-ubuntu-arm64v8:latest`
> 
> only pi4 is arm64.

Not true.  You can install an arm64 image on the Pi3.  Such images were simply not available until the last year or two, but are supported by the architecture.
